src/ExpiringCaches.jl

```diff
@@ -165,9 +165,9 @@ macro cacheable(timeout, func)
     return esc(quote
         const $cacheName = ExpiringCaches.Cache{Tuple{$(argTypes...)}, $returnType}($timeout)
         $internalFunction
-        Base.@__doc__ function $funcName(args...)::$returnType
-            return get!($cacheName, args) do
-                $internalFuncName(args...)
+        Base.@__doc__ function $funcName($(funcArgs...))::$returnType
+            return get!($cacheName, tuple($(funcArgs...))) do
```

Comment:
I think this could just be:
```suggestion
            return get!($cacheName, $(tuple(funcArgs))) do
```

Reply:
Hmm I can't get this to work, gives this error
```Julia
Julia> @cacheable Second(10) function foo(arg1::Int)::Int64
           sleep(5.0)
           return arg1
       end
julia> foo(1)
ERROR: MethodError: no method matching get(::Cache{Tuple{Int64},Int64,Second}, ::Tuple{Array{Any,1}}, ::Symbol)
Closest candidates are:
  get(::REPL.Terminals.TTYTerminal, ::Any, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/Terminals.jl:161
  get(::IdDict{K,V}, ::Any, ::Any) where {K, V} at iddict.jl:86
  get(::Base.Iterators.Pairs, ::Any, ::Any) at iterators.jl:252
  ...
Stacktrace:
 [1] in(::Tuple{Array{Any,1}}, ::Base.KeySet{Tuple{Int64},Cache{Tuple{Int64},Int64,Second}}) at ./abstractdict.jl:65
 [2] haskey(::Cache{Tuple{Int64},Int64,Second}, ::Tuple{Array{Any,1}}) at ./abstractdict.jl:17
 [3] get!(::var"#9#10"{Int64}, ::Cache{Tuple{Int64},Int64,Second}, ::Tuple{Array{Any,1}}) at ./abstractdict.jl:504
 [4] foo(::Int64) at /Users/malmiller/repos/ExpiringCaches.jl/src/ExpiringCaches.jl:169
 [5] top-level scope at REPL[5]:1
```
Although the method error is still good:
```Julia
julia> foo(1.0)
ERROR: MethodError: no method matching foo(::Float64)
Closest candidates are:
  foo(::Int64) at /Users/malmiller/repos/ExpiringCaches.jl/src/ExpiringCaches.jl:168
Stacktrace:
 [1] top-level scope at REPL[6]:1
```

Looking at `@macroexpand`, with `$(tuple(funcArgs)))` we get
```Julia
get!(var"##264", (Any[:(arg1::Int)],))
```
where as with `tuple($(funcArgs...)))` we get
```Julia
get!(var"##265", tuple(arg1::Int))
```
Which makes it look like its not expanding `funcArgs` in the suggested change

Reply:
Oops! I edited my suggestion above; I believe it should be `Tuple` instead of `tuple`

Reply:
Still can't get it to work!
```Julia
julia> foo(1)
ERROR: MethodError: no method matching get(::Cache{Tuple{Int64},Int64,Second}, ::Tuple{Expr}, ::Symbol)
Closest candidates are:
  get(::REPL.Terminals.TTYTerminal, ::Any, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/Terminals.jl:161
  get(::IdDict{K,V}, ::Any, ::Any) where {K, V} at iddict.jl:86
  get(::Base.Iterators.Pairs, ::Any, ::Any) at iterators.jl:252
```
`@macroexpand` gives
```Julia
get!(var"##279", (:(arg1::Int),))
```
I think because `funcArgs` is a `Vector{Expr}`, the `$` needs to be just before the expanded `funcArgs` to ensure the `Expr`s are interpolated

Reply:
Ah yes, you're right. My bad. We need the tuple to happen at runtime to have Exprs evaluated correctly. Ok, let's merge this as-is then.